frontend/src/App.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+
+import {
+  RenderEmbeddings,
+  Navbar,
+  EmbeddingDetails,
+  Footer,
+  Loader,
+} from "./components";
+import { getEmbeddingPoints, getEmbeddingPointDetails } from "./actions";
+import "./index.css";
+
+function App() {
+  const [embeddings, setEmbeddings] = useState<Array<[number, number, number]>>(
+    []
+  );
+
+  const [embeddingPoint, setEmbeddingPoint] = useState<null | number[]>(null);
+
+  const [embeddingDetails, setEmbeddingDetails] = useState<null | {
+    id: number;
+  }>(null);
+
+  const [loadingEmbeddingDetails, setLoadingEmbeddingDetails] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    const embeddings = async () => {
+      const resp = await getEmbeddingPoints();
+      setEmbeddings(resp);
+    };
+    embeddings();
+  }, []);
+
+  const handlePointClicked = async (point: number[]) => {
+    setLoadingEmbeddingDetails(true);
+    setEmbeddingPoint(point);
+
+    const details = await getEmbeddingPointDetails(point);
+
+    setEmbeddingDetails(details);
+    setLoadingEmbeddingDetails(false);
+  };
+
+  const handlePointUnclicked = () => {
```

Comment:
```suggestion
  const handleEmdeddingUnselect = () => {
```
does "handlePointUnselected" better convey this function? "Unclicked" seems a little odd. Also I'd prefer "embedding" over "point", but open to other ideas.